.github/workflows/frontend-ci.yml

```diff
@@ -15,43 +15,79 @@ jobs:
     runs-on: windows-latest
 
     steps:
-    - name: Checkout code
-      uses: actions/checkout@v4
-
-    - name: Cache Node.js modules
-      uses: actions/cache@v4
-      with:
-        path: ~/.npm
-        key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
-        restore-keys: |
-          ${{ runner.os }}-npm-
-
-    - name: Set up Node.js
-      uses: actions/setup-node@v4
-      with:
-        node-version: '16'
-
-    - name: Install dependencies
-      run: |
-        cd frontend
-        npm install
-
-    - name: Lint frontend code
-      run: |
-        cd frontend
-        npm run lint
-
-    - name: Format frontend code with Prettier
-      run: |
-        cd frontend
-        npm run format
-
-    - name: Build frontend
-      run: |
-        cd frontend
-        npm run build
-
-    - name: Run ESLint Security Checks
-      run: |
-        cd frontend
-        npm audit --omit=dev --audit-level=critical || true
\ No newline at end of file
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Cache Node.js modules
+        uses: actions/cache@v4
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-npm-
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: "16"
```

Comment:
hopefully trivial updates like this will go away with the work @ahmedbektic is doing.